### PR TITLE
LPS-194434 | LPS-193944

### PR DESCRIPTION
**Pull request by Artartur**
Related Issues:
https://liferay.atlassian.net/browse/LPS-193944
https://liferay.atlassian.net/browse/LPS-194434

### CODE REVIEW

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/DuplicateObjectDefinitionExternalReferenceCodeExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	@Override
+	protected Problem getProblem(
+		DuplicateObjectDefinitionExternalReferenceCodeException
+			duplicateObjectDefinitionExternalReferenceCodeException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST,
+			_language.get(
+				_acceptLanguage.getPreferredLocale(),
+				"This ERC is already in use. Try another one."));
```
Comment (murilostodolni) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/DuplicateObjectDefinitionExternalReferenceCodeExceptionMapper.java:45]: Hey @Artartur 

We need to add a language key here. I looked in `Language.properties` the key:
`please-enter-a-unique-external-reference-code=Please enter a unique external reference code.`

Is this the expect message? I not found in the LPS.

**Review by murilostodolni — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/DuplicateObjectFolderExternalReferenceCodeExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	@Override
+	protected Problem getProblem(
+		DuplicateObjectFolderExternalReferenceCodeException
+			duplicateObjectFolderExternalReferenceCodeException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST,
+			_language.get(
+				_acceptLanguage.getPreferredLocale(),
+				"This ERC is already in use. Try another one."));
```
[the remark this answers is not in the document]
Comment (Artartur) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/DuplicateObjectFolderExternalReferenceCodeExceptionMapper.java:45]: I've asked reporters to update the tickets with the expected message now and the expected message is `This External Reference Code is already in use. Try another one.`

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:relevant

**Comment by Artartur** [comment]
ci:test:object

**Comment by Artartur** [comment]
ci:test:relevant

**Comment by Artartur** [comment]
ci:test:object

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -18472,6 +18472,7 @@ this-document-type-does-not-belong-to-this-site.-you-may-affect-other-sites-if-y
 this-editor-is-not-optimized-for-mobile-or-400-zooming=This editor is not optimized for mobile or 400% zooming.
 this-event-and-any-future-events-in-the-series-will-be-modified=This event, and any future events in the series, will be modified.
 this-event-does-not-repeat=This event does not repeat.
+this-external-reference-code-is-already-in-use=This External Reference Code is already in use. Try another one.
```
Comment (murilostodolni) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18475]: Could you change this language key to `this-external-reference-code-is-already-in-use-try-another-one`, please?

This key is follow the same pattern [here](https://github.com/liferay/liferay-portal/blob/cb1a32e11e9579147c26ddcca5df579f6d14fc5d/modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties#L18541) and [here](https://github.com/liferay/liferay-portal/blob/cb1a32e11e9579147c26ddcca5df579f6d14fc5d/modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties#L11107).

**Review by murilostodolni — CHANGES_REQUESTED** [review]

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:relevant

**Comment by Artartur** [comment]
ci:test:object

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**